### PR TITLE
Changes for symbolic-map specs part 2: map operations & #buf construct

### PR DESCRIPTION
**Pull request by yzhang90**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -321,7 +321,7 @@ $(proof_dir)/%.test: $(proof_dir)/% split-proof-tests
 	$(TEST) test --backend $(TEST_SYMBOLIC_BACKEND) $<
 
 split-proof-tests: tests/proofs/make.timestamp
-	$(MAKE) -C tests/proofs $@
+	$(MAKE) -C tests/proofs $@ MODE=minimal
```
Comment (daejunpark) [inline Makefile]: goal:
```
$(MAKE) -C tests/proofs $@ MODE=minimal
$(MAKE) -C tests/proofs $@ MODE=k-test
$(MAKE) -C tests/proofs $@ MODE=erc20
```

File: data.md [hunk 2]
```diff
@@ -670,8 +670,8 @@ We are using the polymorphic `Map` sort for these word maps.
 ```k
     syntax Int ::= #lookup ( Map , Int ) [function]
  // -----------------------------------------------
-    rule #lookup( (KEY |-> VAL) M, KEY ) => VAL
-    rule #lookup(               M, KEY ) => 0 requires notBool KEY in_keys(M)
+    rule #lookup( (KEY |-> VAL) M, KEY ) => VAL                               [concrete]
```
Comment (MrChico) [inline data.md]: These `[concrete]` attributes are breaking all of the proofs we are working with. It seems we just have to copy over all of these rules into our lemmas with the `[concrete]` tag removed. I'm kinda confused why you are doing this. Do we need to split up evm-semantics into a symbolic and concrete version?

Comment (denis-bogdanas) [reply data.md]: We had to do the same in verified-smart-contracts. Look at our old projects like erc20/hkg for model. If you use precise gas computation for loops, then take bihu as model.

Comment (daejunpark) [reply data.md]: @MrChico Let us fix this problem.  I think we can find a way without this change.

Comment (ehildenb) [reply data.md]: We will have a follow up PR into K which adds some flags which allow us to specify that we want certain rules to be concrete at `kprove` time.

Comment (MrChico) [reply data.md]: So with that PR we would remove these `[concrete]` flags again from this repo? I guess we'll just wait for that to get merged then

Comment (daejunpark) [reply data.md]: Yep.

**Pull request merged**